src/apyfloatarray.cc

```diff
@@ -563,22 +564,27 @@ APyFloatArray APyFloatArray::operator*(const APyFloatArray& rhs) const
 
             man_t new_man = mx * my;
 
-            // Check carry from multiplication
+            // Check result from multiplication larger than/equal two
             if (new_man & two_before) {
                 tmp_exp++;
                 new_man <<= 1;
-            } else {
-                // Check for subnormal
+            } else if (new_man & one_before) {
+                // Align with longer result
                 if (tmp_exp == 0) {
                     new_man <<= 1;
-                } else if (tmp_exp < 0) {
-                    new_man = (new_man >> (-tmp_exp - 1))
-                        | ((new_man & ((1 << -tmp_exp) - 1)) != 0);
-                    tmp_exp = 0;
                 } else {
-                    // Align with longer result
                     new_man <<= 2;
                 }
+            } else {
+                // One or two of the operands were subnormal
+                new_man <<= std::max(tmp_exp + 1, (std::int64_t)0);
+                tmp_exp = 0;
```

Comment:
I do not think this is correct. This is basically the case where we would like to shift-out parts of the mantissa and by setting tmp_exp to zero this will not happen. Probably the cause of #353.

Reply:
Hm yes that is probably the cause for #353. That line was intended to handle #319 but seems to have broken other things.